src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerDelegate.scala

```diff
@@ -67,8 +67,18 @@ private[tracker] class InstanceTrackerDelegate(
     Await.result(specInstances(appId), instanceTrackerQueryTimeout.duration)
   }
 
-  override def specInstances(appId: PathId)(implicit ec: ExecutionContext): Future[Seq[Instance]] =
-    (instanceTrackerRef ? InstanceTrackerActor.ListBySpec(appId)).mapTo[Seq[Instance]]
+  override def specInstances(appId: PathId)(implicit ec: ExecutionContext): Future[Seq[Instance]] = {
```

Comment:
I think we should somehow reflect in the name of the method that this is a method that won't finish until updates proceed as it's something that I believe most of the marathon code does not want to use. This is kind of a cache for reading all instance data, we should not serialize all read operations...

I think this should NOT be the default query method for this class